lhotse/recipes/iwslt22_ta.py

```diff
@@ -0,0 +1,370 @@
+                    )
+
+        for future in tqdm.tqdm(futures, desc="Processing text", leave=False):
+            result = future.result()
+            if result is None:
+                continue
+            for sup in result:
+                supervisions.append(sup)
+
+        supervisions = deduplicate_supervisions(supervisions)
+        supervisions = SupervisionSet.from_segments(supervisions)
+        recordings = RecordingSet.from_recordings(recordings.values())
+        recordings, supervisions = remove_missing_recordings_and_supervisions(
+            recordings,
+            supervisions,
+        )
+        supervisions = trim_supervisions_to_recordings(recordings, supervisions)
+        validate_recordings_and_supervisions(recordings, supervisions)
+        for split in ("train", "dev", "test1"):
+            sups_ = supervisions.filter(lambda s: s.recording_id in split_files[split])
+            recs_ = recordings.filter(lambda r: r.id in split_files[split])
+            manifests[split] = {"recordings": recs_, "supervisions": sups_}
+
+            if output_dir is not None:
+                output_dir = Path(output_dir)
+                output_dir.mkdir(parents=True, exist_ok=True)
+                sups_.to_file(output_dir / f"iwslt22-ta_recordings_{split}.jsonl.gz")
+                recs_.to_file(output_dir / f"iwslt22-ta_supervisions_{split}.jsonl.gz")
+
+    return manifests
+
+
+def _filename_to_supervisions(
+    p: Path,
+    translations_path: Path,
+    clean: bool,
+    exclude: list,
+    langs: list,
+):
+    supervisions = []
+    date, time, someid, channel = p.with_suffix("").stem.split("_")
+    date_tgt, time_tgt, someid_tgt, channel_tgt = p.with_suffix("").stem.split("_")
+    translations = translations_path.read_text().splitlines()
+    transcripts = p.read_text().splitlines()
+
+    sorted_translations = sorted(translations, key=lambda line: line.split("\t")[0])
+    sorted_transcripts = sorted(transcripts, key=lambda line: line.split("\t")[0])
+
+    for src, tgt in zip(sorted_transcripts, sorted_translations):
+
+        start, end, sid, text = src.rstrip().split("\t")
+        _, _, _, text_tgt = tgt.rstrip().split("\t")
+        start = float(start)
+        end = float(end)
+
+        # Following the IWSLT provided text normalization
+        text = normalize_text(text, "transcript")
+        text_tgt = normalize_text(text_tgt, "translation")
+
+        utt_id = f"{date}_{time}_{someid}_{channel}_{int(100*start):06}"
+        utt_id_tgt = (
+            f"{date_tgt}_{time_tgt}_{someid_tgt}_{channel_tgt}_{int(100*start):06}"
+        )
+
+        assert (
+            utt_id == utt_id_tgt
+        ), f"The loaded source and target files are not sorted properly: {utt_id} {utt_id_tgt}"
+
+        if clean:
+            # Aggressive Tunisian text normalization and cleaning from https://aclanthology.org/2022.iwslt-1.29.pdf

```

Comment:
You can mention this paper in the `help` for the `--clean` (or `--normalize-text`) option, so users can know what kind of normalization will be performed.

Reply:
Done

Reply:
I addressed all the previous comments.  Please let me know If there are any further comments or suggestions.

Reply:
I’ll leave merging up to @desh2608 but I see some tests are failing, looks like you’re missing some import in init py